### PR TITLE
mapping loading with mapping type specified or different method name

### PR DESCRIPTION
**Pull request by ssssarah**
Adresses https://github.com/BlueBrain/nexus-forge/issues/332

I guess the real issue is that a mapping may be a url, txt or a file path, and if it was neither a url (= couldn't request it) or a filepath (= couldn't load the file), then it was assumed to be a txt mapping. But it could have been intended to be a url but malformed, or a filepath but malformed.
Maybe some validation that whatever string is passed is meant to be a text mapping should be done. I've put a hack of a check that it should start with a {, but I'm not even sure this is true of all text mappings.

Update: in checking the validity of the mapping str provided, it brought out that many current mappings in tests are not valid paths, and the failure to load them was silent and did not make the tests fail

### CODE REVIEW

File: kgforge/core/archetypes/store.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def __init__(
         self.token: Optional[str] = token
         self.versioned_id_template: Optional[str] = versioned_id_template
         loaded = (
-            self.mapping.load(file_resource_mapping) if file_resource_mapping else None
+            self.mapping.load_file(file_resource_mapping) if file_resource_mapping else None
```
Comment (crisely09) [inline kgforge/core/archetypes/store.py:125]: here, are you sure it is a file? it could be an url ...

Comment (crisely09) [reply kgforge/core/archetypes/store.py:125]: in fact I am getting the `FileNotFoundError`, for the mapping given in the configuration: `https://raw.githubusercontent.com/BlueBrain/nexus-forge/master/examples/configurations/nexus-store/file-to-resource-mapping.hjson`

Comment (crisely09) [reply kgforge/core/archetypes/store.py:125]: changing it back to `load` makes things work fine

File: tests/specializations/mappings/test_mappings.py [hunk 2]
```diff
@@ -13,3 +13,43 @@
...
+    with exception:
+        mapping = DictionaryMapping.load(source)
+
+
+@pytest.mark.parametrize(
+    "source, mapping_type, exception",
+    [
+        (mapping_path_valid, MappingType.FILE, does_not_raise()),
+        (mapping_url_valid, MappingType.URL, does_not_raise()),
+        ("i", MappingType.URL, pytest.raises(Exception)),
```
Comment (crisely09) [inline tests/specializations/mappings/test_mappings.py:50]: you can add more of the exception cases, to complete the testing (file path being passed as a url, text being fiven as file, url as text, etc.)

File: kgforge/core/archetypes/store.py (hunk 1 above)
Comment (ssssarah) [reply kgforge/core/archetypes/store.py:125]: My mistake

File: tests/specializations/mappings/test_mappings.py (hunk 2 above)
Comment (ssssarah) [reply tests/specializations/mappings/test_mappings.py:50]: I hadn't yet because there was no usage of these, but sure

Comment (ssssarah) [reply tests/specializations/mappings/test_mappings.py:60]: I've added some test cases

**Review by crisely09 — CHANGES_REQUESTED** [review]

**Review by crisely09 — APPROVED** [review]
Looks good

**Pull request merged**